glslc/src/file_compiler.cc

```diff
@@ -62,13 +62,27 @@ bool FileCompiler::CompileShaderFile(const std::string& input_file,
     source_string = string_piece(&(*input_data.begin()),
                                  &(*input_data.begin()) + input_data.size());
   }
-  StageDeducer deducer(input_file);
-  bool compilation_success =
-      Compile(source_string, shader_stage, error_file_name.str(), deducer,
-              glslc::FileIncluder(&include_file_finder_), output_stream,
-              &std::cerr, &total_warnings_, &total_errors_);
 
-  if (!compilation_success && output_stream->fail()) {
+  std::unique_ptr<FileIncluder> includer(new FileIncluder(&include_file_finder_));
+  options_.SetIncluder(std::move(includer));
+
+  shaderc::SpvModule result =
+      CompileGlslToSpv(source_string.begin(), source_string.size(), shader_kind,
```

Comment:
It is not portable to pass `string::iterator` where a `char*` is expected.  Use `.c_str` instead of `.begin`.